src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -1691,18 +1691,19 @@ private void handleSwitch(JCTree switchTree,
                               List<JCCase> cases,
                               BiConsumer<JCCase, Env<AttrContext>> attribCase) {
         Type seltype = attribExpr(selector, env);
+        Type seltypeUnboxed = types.unboxedTypeOrType(seltype);
 
         Env<AttrContext> switchEnv =
             env.dup(switchTree, env.info.dup(env.info.scope.dup()));
 
         try {
             boolean enumSwitch = (seltype.tsym.flags() & Flags.ENUM) != 0;
             boolean stringSwitch = types.isSameType(seltype, syms.stringType);
-            boolean booleanSwitch = types.isSameType(types.unboxedTypeOrType(seltype), syms.booleanType);
+            boolean booleanSwitch = types.isSameType(seltypeUnboxed, syms.booleanType);
             boolean errorEnumSwitch = TreeInfo.isErrorEnumSwitch(selector, cases);
             boolean intSwitch = types.isAssignable(seltype, syms.intType);
             boolean patternSwitch;
-            if (seltype.isPrimitive() && !intSwitch) {
+            if (seltypeUnboxed.isPrimitive() && !intSwitch) {
```

Comment:
Sorry, but this seems wrong. Having a selector type of type `Boolean` is not a preview feature (since JDK 21). I.e. this is completely valid:
```
public class Test {
    public static void main(String... args) {
        Boolean b = true;
        System.err.println(switch (b) { case Boolean bb -> bb; });
    }
}
```

I think there needs to be a preview check in the section that handles constant labels, probably somewhere here:
https://github.com/openjdk/jdk/blob/76f792b55263faf883e54cb879d8609f87164e51/src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java#L1808

Reply:
Ah! My apologies, I totally misunderstood the cause of this issue.

Since this is more complicated than I thought I will defer back to Angelos who is the expert.

Thanks. Closing this PR.